app/views/new_thread.js

```diff
@@ -270,34 +270,45 @@
                     label: 'Invite'
                 }],
                 options: {
-                    onApprove: async () => {
-                        const $input = modal.$modal.find('input');
-                        let phone = $input.val().replace(/[^0-9]/g, '');
-                        if (phone.length < 10) {
-                            modal.$modal.find('.ui.form').addClass('error');
-                            return false;
-                        } else if (phone.length === 10) {
-                            phone = '+1' + phone;
-                        } else if (phone.length === 11) {
-                            phone = '+' + phone;
-                        }
-                        const registered = await F.atlas.findUsers({phone});
-                        if (phone === F.currentUser.attributes.phone) {
-                            const m = new F.ModalView({
-                                icon: 'warning sign red',
-                                header: 'Current phone matches entered phone',
-                                content: 'Please select a different phone number',
-                                actions: [{
-                                    class: 'deny black',
-                                    label: 'Cancel',
-                                }]
-                            });
-                            await m.show();
-                        } else if (registered.length > 0) {
-                            this.suggestFromPhone(registered);
-                        } else {
-                            this.startInvite(phone);
-                        }
+                    onApprove: () => {
+                        (async () => {
```

Comment:
add comment here indicating why we had to inline this async immediate call.  Eg. "onApprove can't be async and we have to return false to indicate that the window shouldn't close automatically."